### PR TITLE
VIH-9744 show deleted users & allow restore

### PR DESCRIPTION
**Pull request by phillipjchambers**
### JIRA link (if applicable) ###
VIH-9744

### Change description ###
* includes deleted users in search results
* indicates deleted users with badge
* allows restore of deleted users & confirmation dialog
* hides delete & edit icons

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 1]
```diff
@@ -76,46 +94,43 @@ export class ManageTeamComponent {
...
-        this.userFormMode = 'edit';
-        this.displayForm();
-    }
+    /*
+        End Add & Edit User
+    */
+
+    /*
+        Delete User
+    */
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:133]: Are these comments required?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 2]
```diff
@@ -76,46 +94,43 @@ export class ManageTeamComponent {
         this.displayMessage = true;
     }
 
-    displayForm() {
-        this.displayMessage = false;
-        this.showForm = true;
+    /*
+        Add & Edit User
+    */
+
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:100]: same as below

**Review by will-craig — APPROVED** [review]
Looks good to me have; have asked about the neccessity of the code comments. But not a blocker

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts (hunk 1 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts]: Removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts (hunk 2 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts]: Removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 3]
```diff
@@ -131,13 +146,61 @@ export class ManageTeamComponent {
...
+        this.userToDelete.deleted = true;
+    }
+
+    /*
+        End Delete User
+    */
+
+    /*
+        Restore User
+    */
```
[the remark this answers is not in the document]
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts]: Removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/services/justice-users.service.ts [hunk 4]
```diff
@@ -1,7 +1,14 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { map, shareReplay } from 'rxjs/operators';
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/services/justice-users.service.ts:3]: I know this wasn't added by your PR but `map` is showing as an unused import - could we remove it?

**Review by oliver-scott — COMMENTED** [review]
Looks good. Just one suggestion, only if it isn't too much trouble

File: AdminWebsite/AdminWebsite/ClientApp/src/app/services/justice-users.service.ts (hunk 4 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/services/justice-users.service.ts]: Thanks, removed

**Review by will-craig — APPROVED** [review]

**Pull request merged**